_im-plugin/index-context.md

```diff
@@ -0,0 +1,170 @@
+            "index": {
+                "codec": "zstd_no_dict",
+                "refresh_interval": "60s",
+                "number_of_shards": "1",
+                "provided_name": "my-metrics-index",
+                "merge": {
+                    "policy": "log_byte_size"
+                },
+                "context": {
+                    "created_version": "1",
+                    "current_version": "1"
+                },
+                ...
+            }
+        },
+        "context": {
+            "name": "metrics",
+            "version": "_latest"
+        }
+    }
+}
+```
+````
+
+#### Considerations
+
+There are few restrictions which are applied when you decide to use `context` param when creating your indices:
+
+1. When you create an index using `context` field, any settings which are declared in context cannot be provided as part of index creation, or dynamic settings updates.
+2. Once set on an index/index-template, context cannot be removed from it.
+
+The above points ensure that at any point, the recommended settings/mappings are always applied on the indices.
+
+
+
+### Create Index Templates
+
+You can also provide the `context` parameter while creating index templates:
+
+````
+```json
+PUT _index_template/my-logs
+{
+    "context": {
+        "name": "logs",
+        "version": "1"
+    },
+    "index_patterns": [
+        "my-logs-*"
+    ]
+}
+```
+{% include copy-curl.html %}
+````
+
+````
+```json
+GET _index_template/my-logs
+```
+{% include copy-curl.html %}
+````
+
+````
+```json
+{
+    "index_templates": [
+        {
+            "name": "my-logs2",
+            "index_template": {
+                "index_patterns": [
+                    "my-logs1-*"
+                ],
+                "context": {
+                    "name": "logs",
+                    "version": "1"
+                }
+            }
+        }
+    ]
+}
+```
+````
+
+
+This will create an index template with the context information as `logs`, and all indices created via this index template will get the metadata provided by the associated component template.
+If there is any overlap between settings/mappings/aliases declared by your template directly and the same coming through the backing component template for the context, the latter gets higher priority during index creation.
+
+## Available Templates for Context
+
+Following templates are available to be used through the `context` parameter as of today:
+
+1. logs
+2. metrics
+3. nginx-logs
+4. amazon-cloudtrail-logs
+5. amazon-elb-logs
+6. amazon-s3-logs
+7. apache-web-logs
+8. k8s-logs
+
+To see more details on what these templates provide, please take a look at the [repository](https://github.com/opensearch-project/opensearch-system-templates/tree/d71e12588e3144ad0fd26745d084ba49e9e08349/src/main/resources/org/opensearch/system/applicationtemplates/v1).
+You can also check the current version of these templates installed on your cluster through `GET /_component_template`
```

Comment:
this will give all the component template? should we provide a filtered. request to show only context component templates

Reply:
What would the filter be @shwetathareja?

Reply:
Yes, I've a PR for that in core. We will be able to take it in the next release.